client/network/src/behaviour.rs

```diff
@@ -351,3 +355,9 @@ impl From<DiscoveryOut> for BehaviourOut {
 		}
 	}
 }
+
+impl From<void::Void> for BehaviourOut {
+	fn from(_e: void::Void) -> Self {
+		Self::None
+	}
+}
```

Comment:
```suggestion
impl From<void::Void> for BehaviourOut {
	fn from(e: void::Void) -> Self {
		void::unreachable(e)
	}
}
```

Why not make use of the impossibility of `Void`, i.e. make use of the fact that `Void` is  _uninhabited_?